src/modules/swap/hooks/useIsSwapEth.ts

```diff
@@ -0,0 +1,10 @@
+import { useDetectNativeToken } from './useDetectNativeToken'
+
+import { useIsWrapOrUnwrap } from '../../trade/hooks/useIsWrapOrUnwrap'
+
+export function useIsSwapEth(): boolean {
+  const { isNativeIn } = useDetectNativeToken()
```

Comment:
It's not required, just a wish for refactoring:

`useDetectNativeToken` is pretty "heavy", because it uses `useTokenBySymbolOrAddress`. If it's possible, will be better to do smth similar to `useIsWrapOrUnwrap` implementation.

Reply:
Will give it a try in another PR